lib/blacklight/abstract_repository.rb

```diff
@@ -0,0 +1,20 @@
+module Blacklight
+  class AbstractRepository
+    attr_accessor :blacklight_config, :blacklight_solr
+
+    # ActiveSupport::Benchmarkable requires a logger method
+    attr_accessor :logger
+
+    include ActiveSupport::Benchmarkable
+
+    def initialize blacklight_config
+      @blacklight_config = blacklight_config
+    end
+
+    protected
+
+      def logger
+        @logger ||= Rails.logger if defined? Rails
```

Comment:
Just making a note for later.. we should probably spin this logic off to `Blacklight.logger` and fallback to our own STDERR logger or something.